File: forum/src/main/java/com/example/forum/controller/UserController.java
package com.example.forum.controller;

import com.example.forum.dto.UserDto;
import com.example.forum.jwt.TokenInfo;
import com.example.forum.service.UserService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RequestMapping("user")
@RestController
public class UserController {
    private final UserService userService;

    public UserController(@Autowired UserService userService) {
        this.userService = userService;
    }

    @PostMapping
    public ResponseEntity<UserDto.UserGetRes> getUser(@Valid @RequestBody UserDto.UserGetReq dto) {
        return ResponseEntity.ok(userService.getUser(dto));
    }

    @PostMapping("/create")
    public ResponseEntity<Long> createUser(@Valid @RequestBody UserDto.SignUp dto) {
        return new ResponseEntity<Long>(userService.createUser(dto), HttpStatus.CREATED);
    }

    @PostMapping("/login")
    public ResponseEntity<TokenInfo> login(@Valid @RequestBody UserDto.UserLoginReq dto) {
        String userId = dto.getUserId();
        String password = dto.getPassword();
        return ResponseEntity.ok(userService.login(userId, password));
    }

    @PostMapping("/memo")
    public ResponseEntity<Long> changeMemo(@Valid @RequestBody UserDto.ChangeMemoReq dto) {
        return ResponseEntity.ok(userService.changeMemo(dto));
    }
}

File: migration-to-kotlin/src/main/kotlin/com/example/practice/controller/UserController.kt
package com.example.practice.controller

import com.example.practice.common.PageRequest
import com.example.practice.dto.UserDto
import com.example.practice.service.UserService
import jakarta.validation.Valid
import org.springframework.data.domain.Page
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/user")
@RestController
class UserController(private val userService: UserService) {
    @PostMapping
    @ResponseStatus(HttpStatus.OK)
    internal fun createUser(@RequestBody @Valid dto : UserDto.SignUp) : Long {
        return userService.createUser(dto)
    }

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    internal fun findUser(@RequestParam("userId") userId : String) : UserDto.Res {
        return userService.findById(userId)
    }

    @GetMapping("/list")
    @ResponseStatus(HttpStatus.OK)
    internal fun findUsers(pageRequest: PageRequest) : Page<UserDto.Res> {
        return userService.findUsers(pageRequest.of())
    }
}